library/core/src/net/socket_addr.rs

```diff
@@ -96,7 +94,7 @@ pub struct SocketAddrV4 {
 /// assert_eq!(socket.ip(), &Ipv6Addr::new(0x2001, 0xdb8, 0, 0, 0, 0, 0, 1));
 /// assert_eq!(socket.port(), 8080);
 /// ```
-#[derive(Copy, Clone, Eq, PartialEq)]
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
```

Comment:
Can you add some tests for `SocketAddrV6`s with different scope IDs and flow info?

Reply:
Does [5c13c69](https://github.com/rust-lang/rust/pull/116714/commits/5c13c69f63dccdc637d28b403ae1fb687e0a5143) look good to you?

Reply:
Yep, thanks!